content/posts/2020-07-15-tweeting-from-pull-requests/index.md

```diff
@@ -0,0 +1,45 @@
+
+## The problem
+
+Twitter is a [microblogging](https://en.wikipedia.org/wiki/Microblogging) platform so I thought: "Why not having a similar experience from regular blogs where multiple authors write multiple articles?". Well, that would be nice. But unfortunately, different from regular blogs, we can only post content from a single account.
+
+The easiest solution would be sharing the credentials, some may think. That would work but that only wouldn’t be enough. Collective ownership is also about making sure decisions are made and agreed upon. We would need at least people reviewing each other’s content.
+
+So thinking a bit more, what I envisioned was a flow that would allow us to:
+
+1. have someone proposing a new tweet
+2. notify others when there’s new content to be reviewed
+3. have others reviewing and approving it
+4. publish tweet when content is approved
+
+![./diagram.png](diagram.png)
+
+## The solution
+
+Because these reviewers are mostly engineers, the closer this whole process is from their natural habitat, the better. We use GitHub and Slack all day so the solution would definitely be somewhere in there.
+
+I have used [Danger](https://danger.systems/js/) and [Peril](https://github.com/danger/peril) quite a lot in the past. It’s a great tool to automate many of the workflows we have, especially when it comes to review of Pull Requests. In fact, Loadsmart has its own set of [settings](https://github.com/loadsmart/peril-settings) that are used across many of our repositories on GitHub. But this time I decided to give GitHub actions a try.
+
+Surprisingly, most of the things I was planning to implement were already done by [Gregor Martynus](https://github.com/gr2m) in their [twitter-together](https://github.com/gr2m/twitter-together) project. From the steps of the diagram above, the Action takes care of 1, 2 and 4. The only step missing here is the Slack notification.
+
+There’s a few Actions to interact with Slack in the marketplace. They are all very similar to each other and it seems to me that the same problem was solved multiple times (thinking out loud, it would be good if GitHub had a way to make Actions "official"). So I decided to implement this step myself. I would have to create a custom action in the end to glue all the work together (passing to Slack the content of the tweet in the PR) so this would be only an extra step. By the way, their [Block Kit API](https://api.slack.com/block-kit) is very flexible and you can build interesting custom notifications:
```

Comment:
It is not clear for me, reading the text, why you decided to create it by itself. Reading the first sentence, I would guess you would pick any of the existent solutions. Otherwise, you were solving the same problem one more time.

Maybe there were other reasons that pointed you to decided to do it by yourself. Perhaps because there was no official one or one that was the most adopted. Or you did not trust the ones you checked. I don't know, but I believe we can talk more a bit about it here and make it more explicit.

What do you think?

Reply:
Agreed. Will explain a bit better.